### DISCUSSION TITLE
Does events have access to the userid of the request?

### DISCUSSION THREAD

**Discussion by rschoultz** [pocketbase/pocketbase#227] [Q&A] [answered]
I want to determine the userId of the user that has invoked a request, and access that from e.g.

app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
    // 1. Find out userId of the requestor, for the purpose of logging and maybe even rego evaluations.
    // 2. Store the userId on the record, determined by the server-side, for later use in the API Rules.
}

I started down the path of parsing the JWT of header to get the id of the user, but as that is already determined there might be some other way?

**Comment by ganigeorgiev** [comment, marked as answer]
This is not documented at the moment (it will be added in the near future), but Yes - you can access the authenticated user or admin model by getting it from the request context. For example:
```go
import (
	"log"
	
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/models"
)

...

app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
    user, _ := e.HttpContext.Get(apis.ContextUserKey).(*models.User)
    if user != nil {
        // authenticated user
        log.Println(user.Id)
    }

    return nil
})
```

**Reply by rschoultz** [reply]
Perfect. Thanks!

**Reply by kigiri** [reply]
Ahh was looking for this, would be nice to include in the example code I think !
